sources/osgShadow/shaders/tapPCF.glsl

```diff
@@ -2,12 +2,44 @@
 
 #pragma include "shadowLinearSoft.glsl"
 
+float hash1D(float n) { return fract(sin(n) * 1e4); }
+float hash2D(vec2 p) { return fract(1e4 * sin(17.0 * p.x + p.y * 0.1) * (0.1 + abs(sin(p.y * 13.0 + p.x)))); }
+
+
+float noise2D(vec2 x) {
```

Comment:
is noise2D general ? if yes it should go in a different glsl file, else it should be prefixed by something like shadow something